classes/schedule.php

```diff
@@ -51,19 +51,55 @@ public function __construct( $wpdb, $version, $login_credentials, $slug, $wordpr
 		$this->logging             = $logging;
 		$this->schedulable_classes = $schedulable_classes;
 
-		$this->identifier = $this->schedule_name;
+		//$this->identifier = $this->schedule_name;
 
-		$this->add_filters();
-		add_action( $this->schedule_name, array( $this, 'maybe_handle' ) ); // run the handle method
+		//$this->add_actions();
+		//add_action( $this->schedule_name, array( $this, 'maybe_handle' ) ); // run the handle method
+		add_action( 'init', array( $this, 'add_actions' ) );
 
 	}
 
 	/**
-	* Create the filters we need to run
+	* Create the actions to run
 	*
 	*/
-	public function add_filters() {
-		add_filter( 'cron_schedules', array( $this, 'set_schedule_frequency' ) );
+	public function add_actions() {
+
+		// create a recurring action for each schedulable item
+		foreach ( $this->schedulable_classes as $key => $value ) {
+
+			$schedule_number = absint( get_option( 'object_sync_for_salesforce_' . $key . '_schedule_number', '' ) );
+			$schedule_unit   = get_option( 'object_sync_for_salesforce_' . $key . '_schedule_unit', '' );
+
+			switch ( $schedule_unit ) {
+				case 'minutes':
+					$seconds = 60;
+					break;
+				case 'hours':
+					$seconds = 3600;
+					break;
+				case 'days':
+					$seconds = 86400;
+					break;
+				default:
+					$seconds = 0;
+			}
+
+			$key = $schedule_unit . '_' . $schedule_number;
+
+			if ( isset( $value['initializer'] ) ) {
+
+				$time      = current_time( 'timestamp' );
+				//$hook      = $value['class']->$value['initializer'];
+				$hook = $value['class'] . '->' . $value['initializer'];
+				//echo 'hook is ' . $hook;
+				//$action_id = as_schedule_recurring_action( $time, $seconds, $hook );
+
+				//error_log( 'time is ' . current_time( 'timestamp' ) . ' and seconds is ' . $seconds . ' and class is ' . $value['class'] . ' and init is ' . $value['initializer'] );
+
+				//as_schedule_recurring_action( current_time( 'timestamp' ), $seconds, array( $value['class'], $value['initializer'] ), array(), '' );
```

Comment:
Looks like the `add_actions` function may not do anything at the moment? I don't see anything returned. Nor any function calls that would have a side effect.